HW5/func.cpp

```diff
@@ -0,0 +1,123 @@
+#include "func.h"
+#include "tree.h"
+#include <cmath>
+#include <functional>
+#include <iostream>
+
+void TaskOne() {
+    BinaryTree* tree{};
+    std::function<void()> tasks[] = { [&tree]() {
+                                       int data;
+                                       std::cout << "Enter value to add: \n";
+                                       std::cin >> data;
+                                       InputElement(tree, data);
+                                     },
+                                     [&tree]() { PrintTree(tree); },
+                                     [&tree]() {
+                                       int data;
+                                       std::cout << "Enter value to delete: \n";
+                                       std::cin >> data;
+                                       DeleteElement(tree, data);
+                                       DeleteElement(tree, data);
+                                     }
+
+    };
+
+    while (true) {
+        std::cout << "Choose your actions with BTS: \n";
+
+        std::cout << "1. Insert element\n";
+        std::cout << "2. Print tree\n";
+        std::cout << "3. Delete element\n";
+        std::cout << "0. Exit\n";
+        unsigned int choise{};
+        std::cin >> choise;
+        if (choise > 3) {
+            std::cout << "Incorrect! Try again\n";
+            continue;
+        }
+        if (choise == 0) {
+            break;
+        }
+        tasks[choise - 1]();
+    }
+}
+
+static bool check_string(std::string str) {
+    for (int n{ 0 }; n < str.size(); ++n) {
+        if (str[n] > 57 || str[n] < 48)
```

Comment:
Лучше писать не коды символов из ASCII, а сами символы:
```suggestion
        if (str[n] > '9' || str[n] < '0')
```

Reply:
Вот это точно! Все время забываю